lib/streamaggr/streamaggr.go

```diff
@@ -600,17 +613,17 @@ func newAggregator(cfg *Config, pushFunc PushFunc, ms *metrics.Set, opts *Option
 
 	a.wg.Add(1)
 	go func() {
-		a.runFlusher(pushFunc, alignFlushToInterval, skipIncompleteFlush, interval, dedupInterval)
+		a.runFlusher(pushFunc, alignFlushToInterval, skipIncompleteFlush, interval, dedupInterval, ignoreFirstIntervals)
 		a.wg.Done()
 	}()
 
 	return a, nil
 }
 
-func (a *aggregator) runFlusher(pushFunc PushFunc, alignFlushToInterval, skipIncompleteFlush bool, interval, dedupInterval time.Duration) {
-	alignedSleep := func(d time.Duration) {
+func (a *aggregator) runFlusher(pushFunc PushFunc, alignFlushToInterval, skipIncompleteFlush bool, interval, dedupInterval time.Duration, ignoreFirstIntervals int) {
+	alignedSleep := func(d time.Duration) bool {
```

Comment:
Is this supposed to prevent from starting the ticker if aggregator was closed?
Do I understand this change correct?

Reply:
to avoid unnecessary flush operation even if it has nil push function

Reply:
As I understood from the code, `alignedSleep` returns `false` only if `a.stopCh` is closed. This may happen only on aggregator shutdown just immediately after the `runFlusher` call, e.g. just after aggregator initialization. In this case the aggregator state is empty, so it is OK to call `flush` on it with `nil` pushFunc, since it doesn't spend any measurable CPU time on flushing an empty state.

Is my understanding correct?